Terminal.Gui/Views/Menu/MenuBar.cs

```diff
@@ -1339,14 +1339,14 @@ private bool Select (int index)
     #region Mouse Handling
 
     /// <inheritdoc/>
-    public override bool OnLeave (View view)
+    internal void LostFocus (View view)
     {
-        if (((!(view is MenuBar) && !(view is Menu)) || (!(view is MenuBar) && !(view is Menu) && _openMenu is { })) && !_isCleaning && !_reopen)
+        if (((!(view is MenuBar) && !(view is Menu))) && !_isCleaning && !_reopen)
```

Comment:
You can make the boolean expression here soooooo much easier to grok by using pattern syntax:

```csharp
if(view is not MenuBar and not Menu && !_isCleaning && !_reopen)
```

Or, by deMorgan, its logical equivalent would be:

```csharp
if ((view is MenuBar or Menu || _isCleaning || _reOpen) is false)

// or

if (!(view is MenuBar or Menu || _isCleaning || _reOpen)) //But I think it's easy to miss the bang.
```

Reply:
Just following up on that one.

Any time you've got a bunch of "not" in an expression, especially if it leads to a ton of parentheses or non-obvious operator precedence, consider deMorgan.

It's also yet another thing R# can just do for you effortlessly, reversibly, and non-destructively. 🥳 

Place the caret on one of the && or || operators and then use the "`Replace && with ||`" (or the other way around) option in the hammer/lightbulb menu. That does deMorgan for you. ***DON'T*** pick the option that says "`Negate && (change semantics)`" because that one will do exactly what it says - invert the actual logical result AND deMorgan it. You probably don't want that.